db/db_impl/db_impl_open.cc

```diff
@@ -1202,24 +1206,39 @@ Status DBImpl::RecoverLogFiles(const std::vector<uint64_t>& wal_numbers,
 
       // We create a new batch and initialize with a valid prot_info_ to store
       // the data checksums
-      WriteBatch batch;
+      std::unique_ptr<WriteBatch> batch(new WriteBatch());
```

Comment:
Separating the input and output argument is a great idea! Besides the saved heap allocation, I think the API is more clear now, making the unit tests more explicit. And also the output argument is only needed for the `kReconcileInconsistency` mode, it can be made an optional argument now.